src/config.c

```diff
@@ -521,6 +532,7 @@ bool Config_Option_set(Config_Option_Field field, RedisModuleString *val) {
 				if (!_Config_ParseInteger(val, &query_mem_capacity)) return false;
 
 				Config_query_mem_capacity_set(query_mem_capacity);
+				_on_config_change(field, &query_mem_capacity);
```

Comment:
Remove all calls to `_on_config_change`
call `_on_config_change` just before returning `true`